### ISSUE TITLE
BUG: `renderer.lua:1082:` Key not found: `neo_tree_position`

### ISSUE THREAD

**Issue by Sleepful** [nvim-neo-tree/neo-tree.nvim#1046]
### Did you check docs and existing issues?

- [X] I have read all the docs.
- [X] I have searched the existing issues.
- [X] I have searched the existing discussions.

### Neovim Version (nvim -v)

NVIM v0.10.0-dev-445+g700cab0068

### Operating System / Version

macos ventura

### Describe the Bug

Getting this weird thing sometimes, mostly when I recently opened vim and I jump to some file through telescope. Not sure how to replicate. It get it more often when I start neovim as `neovim .` 
```
[Neo-tree ERROR] debounce  neo-tree-follow  error:  ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1082: Key not found:
 neo_tree_position
[Neo-tree ERROR] debounce  document_symbols_refresh  error:  ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1082: Key n
ot found: neo_tree_position
[Neo-tree ERROR] Error in event handler for event vim_buffer_added[filesystem.vim_buffer_added]: ...are/nvim/lazy/neo-tree.nvim/lua/
neo-tree/ui/renderer.lua:1082: Key not found: neo_tree_position
[Neo-tree ERROR] Error in event handler for event vim_buffer_deleted[filesystem.vim_buffer_deleted]: ...are/nvim/lazy/neo-tree.nvim/
lua/neo-tree/ui/renderer.lua:1082: Key not found: neo_tree_position
[Neo-tree ERROR] Error in event handler for event vim_buffer_modified_set[filesystem.vim_buffer_modified_set]: ...are/nvim/lazy/neo-
tree.nvim/lua/neo-tree/ui/renderer.lua:1082: Key not found: neo_tree_position
[Neo-tree ERROR] debounce  document_symbols_refresh  error:  ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1082: Key n
ot found: neo_tree_position
E5108: Error executing lua: ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1082: Key not found: neo_tree_position
stack traceback:
        [C]: in function 'nvim_buf_get_var'
        ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1082: in function 'window_exists'
        ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:120: in function 'close'
        ...re/nvim/lazy/neo-tree.nvim/lua/neo-tree/command/init.lua:85: in function 'execute'
        ...l/share/nvim/lazy/LazyVim/lua/lazyvim/plugins/editor.lua:13: in function <...l/share/nvim/lazy/LazyVim/lua/lazyvim/plugin
s/editor.lua:12>

```

### Your Configuration

```Lua
the one that ships with lazyvim.org https://www.lazyvim.org/plugins/editor#neo-treenvim
```

**Comment by Sleepful** [comment]
Specifically I see this issue every time I open `neovim .` and _then_ use [telekasten plugin](https://github.com/renerocksai/telekasten.nvim) "goto_today" to create a new file.

**Comment by tmpm697** [comment]
I got this same error when switch session via telescope.nvim with https://github.com/olimorris/persisted.nvim
```
Error executing vim.schedule lua callback: ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1076: Key not found: neo_tree_position
stack traceback:
	[C]: in function 'nvim_buf_get_var'
	...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1076: in function 'window_exists'
	...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1145: in function 'draw'
	...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1285: in function 'show_nodes'
	...ree.nvim/lua/neo-tree/sources/filesystem/lib/fs_scan.lua:95: in function 'render_context'
	...ree.nvim/lua/neo-tree/sources/filesystem/lib/fs_scan.lua:143: in function <...ree.nvim/lua/neo-tree/sources/filesystem/lib/fs_scan.lua:142>
Error executing vim.schedule lua callback: ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1076: Key not found: neo_tree_position
stack traceback:
	[C]: in function 'nvim_buf_get_var'
	...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1076: in function 'window_exists'
	...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1145: in function 'draw'
	...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1285: in function 'show_nodes'
	...ree.nvim/lua/neo-tree/sources/filesystem/lib/fs_scan.lua:95: in function 'render_context'
	...ree.nvim/lua/neo-tree/sources/filesystem/lib/fs_scan.lua:143: in function <...ree.nvim/lua/neo-tree/sources/filesystem/lib/fs_scan.lua:142>
Error executing vim.schedule lua callback: ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1076: Key not found: neo_tree_position
stack traceback:
	[C]: in function 'nvim_buf_get_var'
	...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1076: in function 'window_exists'
	...re/nvim/lazy/neo-tree.nvim/lua/neo-tree/command/init.lua:147: in function 'do_show_or_focus'
	...re/nvim/lazy/neo-tree.nvim/lua/neo-tree/command/init.lua:202: in function 'callback'
	...share/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/inputs.lua:82: in function 'on_submit'
	...c/.local/share/nvim/lazy/nui.nvim/lua/nui/input/init.lua:152: in function <...c/.local/share/nvim/lazy/nui.nvim/lua/nui/input/init.lua:139>
Error executing vim.schedule lua callback: ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1076: Key not found: neo_tree_position
stack traceback:
	[C]: in function 'nvim_buf_get_var'
	...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1076: in function 'window_exists'
	...re/nvim/lazy/neo-tree.nvim/lua/neo-tree/command/init.lua:147: in function 'do_show_or_focus'
	...re/nvim/lazy/neo-tree.nvim/lua/neo-tree/command/init.lua:202: in function 'callback'
	...share/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/inputs.lua:82: in function 'on_submit'
	...c/.local/share/nvim/lazy/nui.nvim/lua/nui/input/init.lua:152: in function <...c/.local/share/nvim/lazy/nui.nvim/lua/nui/input/init.lua:139>
/home/user/path/to/a/folder
Error executing vim.schedule lua callback: ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1076: Key not found: neo_tree_position
stack traceback:
	[C]: in function 'nvim_buf_get_var'
	...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1076: in function 'window_exists'
	...re/nvim/lazy/neo-tree.nvim/lua/neo-tree/command/init.lua:147: in function 'do_show_or_focus'
	...re/nvim/lazy/neo-tree.nvim/lua/neo-tree/command/init.lua:202: in function 'callback'
	...share/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/inputs.lua:82: in function 'on_submit'
	...c/.local/share/nvim/lazy/nui.nvim/lua/nui/input/init.lua:152: in function <...c/.local/share/nvim/lazy/nui.nvim/lua/nui/input/init.lua:139>
```
steps:
1. vi /path/to/file
2. switch session via telescope
3. error above

I have persisted.nvim setup that close/re-open neo-tree window when save/restore session.
it works normally except if i do step 1 above then switch

normal work mean, if i just issue `nvim` command and then do switching sessions with telescope (not vi /path/to/file)

**Comment by OlivierBBB** [comment]
I get the
```bash
Key not found: neo_tree_position
```
message whenever I navigate to a new path with the `:e go/to/this/file/path` command. Afterwards Neotree stops working correctly. For instance I have `:Neotree toggle` set to `<C-n>` - running this command after the `:e` command produces error messages.

**Comment by Tacsiazuma** [comment]
Wrapped the `vim.api.nvim_buf_get_var` on line 1096 in a `pcall` as a temporary workaround. Don't know where that buffer variable gets lost.

**Comment by Tacsiazuma** [comment]
@cseickel when can we expect this fix on 2.x or 3.x?

**Comment by cseickel** [comment]
@Tacsiazuma this has just been released in [v3.6](https://github.com/nvim-neo-tree/neo-tree.nvim/releases/tag/3.6). 

It won't be backported to 2.x.
